client/layout/masterbar/partner-signup.tsx

```diff
@@ -0,0 +1,27 @@
+import { Gridicon } from '@automattic/components';
+import { useLocalizeUrl } from '@automattic/i18n-utils';
+import { createInterpolateElement } from '@wordpress/element';
+import { useI18n } from '@wordpress/react-i18n';
+import WordPressLogo from 'calypso/components/wordpress-logo';
+
+import './partner-signup.scss';
+
+export default function PartnerSignupMasterbar() {
```

Comment:
I see why we need to create a different Masterbar from the design, but this seems like a lot of code for one link. Is that because of the hackiness? Is it worth doing?

Reply:
Renamed to MasterbarLogin, attempted to re-use `Masterbar` but it had too much baggage, cleaned the new component up a bit its quite a small component at this point. The goBackUrl prop should be moved to child elements using `Masterbar/Item`'s but I couldn't clear the styling on those buttons and opted for something simple for now. I think there needs to be a wider discussion about the masterbar+layout components and the GB reskin as it's not easy to integrate the two.